core/input/input.cpp

```diff
@@ -731,6 +735,43 @@ Vector3 Input::get_gyroscope() const {
 	return gyroscope;
 }
 
+Vector2 Input::get_joy_touchpad_finger_position(int p_device, int p_touchpad, int p_finger) const {
+	_THREAD_SAFE_METHOD_
+	if (!joy_touch.has(p_device) || !joy_touch[p_device].touchpad_fingers.has(p_touchpad) || !joy_touch[p_device].touchpad_fingers[p_touchpad].has(p_finger)) {
+		return Vector2();
+	}
+	return joy_touch[p_device].touchpad_fingers[p_touchpad][p_finger].position;
+}
+
+float Input::get_joy_touchpad_finger_pressure(int p_device, int p_touchpad, int p_finger) const {
+	_THREAD_SAFE_METHOD_
+	if (!joy_touch.has(p_device) || !joy_touch[p_device].touchpad_fingers.has(p_touchpad) || !joy_touch[p_device].touchpad_fingers[p_touchpad].has(p_finger)) {
+		return 0.0f;
+	}
+	return joy_touch[p_device].touchpad_fingers[p_touchpad][p_finger].pressure;
+}
+
+TypedArray<int> Input::get_joy_touchpad_fingers(int p_device, int p_touchpad) const {
+	_THREAD_SAFE_METHOD_
+	if (!joy_touch.has(p_device) || !joy_touch[p_device].touchpad_fingers.has(p_touchpad)) {
+		return TypedArray<int>();
+	}
+
+	TypedArray<int> result;
+	for (auto &i : joy_touch[p_device].touchpad_fingers[p_touchpad]) {
```

Comment:
`auto` is [not allowed](https://contributing.godotengine.org/en/latest/engine/guidelines/cpp_usage_guidelines.html#auto-keyword)